form/src/App.js

```diff
@@ -1,19 +1,29 @@
-import React from "react";
+import React, {useEffect} from "react";
 import { render } from "react-dom";
 import Form from "@rjsf/bootstrap-4";
 import Modal from "react-bootstrap/Modal";
 import Navbar from "react-bootstrap/Navbar";
 import Nav from "react-bootstrap/Nav";
-import Select from "react-select";
+//import Select from "react-select";
 import logo from "./images/logo.png";
 import "./App.css"
 
 const yaml = require("js-yaml");
 
 function App() {
-  const [isOpen, setIsOpen] = React.useState(false);
+  const [isOpen, setIsOpen] = React.useState(false); 
   const [content, setContent] = React.useState("Transitioning...");
   const [formData, setFormData] = React.useState({});
+  const [url, setUrl] = React.useState("");
+  const [filename, setFilename] = React.useState("");
+
+  useEffect(() => {
+    var data = new Blob([content]);
+    setUrl(URL.createObjectURL(data));
+    const name = formData.vendor+formData.model+formData.variant+".yaml"
```

Comment:
This is mostly fine, however please modify the name to follow the `device_id`. It's defined by the following pseudo code:

```
filename = "${ vendor.toLowerCase() }_${ model.toLowerCase()"
if (variant != "") {
  filename = filename + '-' + variant.toLowerCase()
}
filename = filename.replace(' ', '-')
filename = filename.replaceNonAsciiChars()
```

So, all lowercase, no spaces, only ASCII chars and attach the `variant` only if defined. Currently the filename would become something like `TP-Link Router 5000undefined.yml` while it should be `tp-link_router-5000.yaml` and `Foobar Test 4000 v3` should become `foobar_test-4000-v3`.